### ISSUE TITLE
Segfault on exit with version 4.2.1 and Python 3.12

### ISSUE THREAD

**Issue by nilason** [wxWidgets/Phoenix#2479]
**Operating system**: macOS 13.6.1 22G313 arm64
**wxPython version & source**: 4.2.1 (pypi and self-built)         
**Python version & source**: Python 3.12.0 (official Mac dist and MacPorts' built)            

**Description of the problem**:

Exiting/quitting a Python 3.12 session with wxPython loaded causes segmentation fault:

```
Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000010
Exception Codes:       0x0000000000000001, 0x0000000000000010

Termination Reason:    Namespace SIGNAL, Code 11 Segmentation fault: 11
Terminating Process:   exc handler [64502]

VM Region Info: 0x10 is not in any region.  Bytes before following region: 105553518919664
      REGION TYPE                    START - END         [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      UNUSED SPACE AT START
--->  
      MALLOC_NANO (reserved)   600018000000-600020000000 [128.0M] rw-/rwx SM=NUL  ...(unallocated)

Thread 0 Crashed::  Dispatch queue: com.apple.main-thread
0   Python                        	       0x1016b823c _PyObject_Free + 36
1   siplib.cpython-312-darwin.so  	       0x10159f570 finalise + 132 (siplib.c:2064)
2   Python                        	       0x1017a020c Py_FinalizeEx + 452
3   Python                        	       0x1017a1780 Py_Exit + 20
4   Python                        	       0x1017a8b50 handle_system_exit + 44
5   Python                        	       0x1017a8004 _PyErr_PrintEx + 56
6   Python                        	       0x1017a7d30 PyRun_SimpleStringFlags + 144
7   Python                        	       0x1017cad48 pymain_run_command + 148
8   Python                        	       0x1017ca6f0 Py_RunMain + 244
9   Python                        	       0x1017cab00 Py_BytesMain + 40
10  dyld                          	       0x198fc3f28 start + 2236
```

Above log extract is from self-built wxPython with MacPorts' Python (with SIP 6.7.12), but nearly identical with official Python dist with pypi version of wxPython.

The same procedure works as expected with Python <=3.11.

<details>
  <summary>Code Example (click to expand)</summary>

```python
import wx
exit()
```
</details>

[macports_build.log.txt](https://github.com/wxWidgets/Phoenix/files/13347321/macports_build.log.txt)
[official_dist.log.txt](https://github.com/wxWidgets/Phoenix/files/13347322/official_dist.log.txt)

**Comment by nilason** [comment]
> Possible duplicate of #2455.

Sure looks like it. Perhaps better to keep this ticket open till next release, to avoid new reports.

**Comment by mojca** [comment]
Can someone please try to advise me how to generate the sip files?
I tried running
```bash
python3.12 ./build.py dox
python3.12 ./build.py etg
# I had to manually change the path to sip
python3.12 ./build.py sip
```
but this only ended up deleting existing files from the release source tarball.

Are there any "snapshot source releases" available anywhere? I only see binary wheels.

**Comment by swt2c** [comment]
Fixed in 4.2.2.
